### ISSUE TITLE
v2.12 breaks for IPv4 only setup

### ISSUE THREAD

**Issue by kai-66degrees** [terraform-aws-modules/terraform-aws-vpn-gateway#68]
## Description

I was using this module earlier and it was working perfectly however upgrading to v2.12 adds an error related to IPv6 configuration, which I am not using.

Rolling back to v2.11.1 solves the issue.

## Error

Error: Missing required argument
│ 
│   with module.aws-gcp-vpn-tunnels.module.vpn_gateway.aws_vpn_connection.tunnel_preshared[0],
│   on .terraform/modules/aws-gcp-vpn-tunnels.vpn_gateway/main.tf line 304, in resource "aws_vpn_connection" "tunnel_preshared":
│  304:   local_ipv6_network_cidr  = var.local_ipv6_network_cidr
│ 
│ "local_ipv6_network_cidr": all of
│ `local_ipv6_network_cidr,transit_gateway_id` must be specified

**Comment by ZeroDeth** [comment]
same issue: 
```
Error: error modifying EC2 VPN Connection (vpn-07bcd52b7e0637789) connection options: InvalidParameterValue: The LocalIpv6NetworkCidr parameter cannot be specified for vpn of addressing type IPv4 status code: 400, request id: 7abd4e79-b941-4085-bb57-c3e1c967ad51
with module.vpn_gateway.aws_vpn_connection.tunnel_preshared[0]
on .terraform/modules/vpn_gateway/main.tf line 240, in resource "aws_vpn_connection" "tunnel_preshared":
resource "aws_vpn_connection" "tunnel_preshared" {
```

**Comment by antonbabenko** [comment]
Could someone provide a piece of configuration that breaks?

**Comment by kai-66degrees** [comment]
module "vpn_gateway" {
  source  = "terraform-aws-modules/vpn-gateway/aws"
  version = "2.11.1"

  vpc_id                        = var.vpc_id
  vpn_gateway_id                = aws_vpn_gateway.vpn_gw.id
  customer_gateway_id           = aws_customer_gateway.main.id
  create_vpn_connection         = true
  create_vpn_gateway_attachment = true

  vpc_subnet_route_table_count = var.route_table_count
  vpc_subnet_route_table_ids   = var.route_table_id

  tunnel1_inside_cidr = var.custom_tunnel1_inside_cidr
  tunnel2_inside_cidr = var.custom_tunnel2_inside_cidr

  tunnel1_preshared_key = var.custom_tunnel1_preshared_key
  tunnel2_preshared_key = var.custom_tunnel2_preshared_key

  local_ipv4_network_cidr  = var.local_ipv4_network_cidr
  remote_ipv4_network_cidr = var.remote_ipv4_network_cidr

  tags = var.tags

}

**Comment by rvega-devopsgroup** [comment]
The following code works on version 2.11.1 but doesn't work on version 2.12.0

```
module "vpn_gateway" {
  source  = "terraform-aws-modules/vpn-gateway/aws"
  version = "2.12.0"

  vpc_id                                                              = module.vpc.vpc_id
  vpn_gateway_id                                              = module.vpc.vgw_id
  customer_gateway_id                                     = module.vpc.cgw_ids[0]
  vpn_connection_static_routes_only              = true
  vpn_connection_static_routes_destinations = [module.vpn_vpc.vpc_cidr_block]

  # precalculated length of module variable vpc_subnet_route_table_ids
  vpc_subnet_route_table_count = 3
  vpc_subnet_route_table_ids     = module.vpc.private_route_table_ids

  tags = {
    Name = var.name
  }
}
```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 2.12.1 :tada:

**Comment by antonbabenko** [comment]
Thank you guys for providing code samples. Please try version 2.12.1, where this issue was fixed.
